q2_feature_classifier/_consensus_assignment.py

```diff
@@ -0,0 +1,243 @@
+        consensus = _compute_consensus_annotations(
+            obs_taxa, min_consensus=min_consensus,
+            unassignable_label=unassignable_label)
+        # If output_no_hits=True, find/add query IDs missing from consensus
+        if output_no_hits is True:
+            consensus = _output_no_hits(
+                consensus, exp_seq_ids, unassignable_label)
+        # if there are no consensus results (no hits for any query), default
+        # to empty result file (must create empty dict or fail on pd error)
+        if not consensus:
+            consensus = {'': ('', '')}
+        result = pd.DataFrame.from_dict(consensus, 'index')
+        result.index.name = 'Feature ID'
+        result.columns = ['Taxon', 'Confidence']
+        return result
+
+
+def _validate_params(min_id, maxaccepts, min_consensus):
+    if min_id > 1.0 or min_id <= 0.0:
+        raise ValueError('min_id must be > 0 and <= 1')
+    if maxaccepts < 1:
+        raise ValueError('maxaccepts must be >= 1')
+    if min_consensus > 1.0 or min_consensus <= 0.5:
+        raise ValueError('min_consensus must be > 0.51 and <= 1.0')
+
+
+def _output_no_hits(obs_taxa, exp_seq_ids,
+                    unassignable_label=_get_default_unassignable_label()):
+    '''If a query ID has no hits, report as unassigned.'''
+    # extract list of query sequence IDs
+    exp = _open_list_or_file(exp_seq_ids)
+    exp = [line.strip('>') for line in exp if line.startswith('>')]
+
+    # iterate over expected IDs, add as unassigned to obs_taxa if missing
+    for _id in exp:
+        if _id not in obs_taxa:
+            obs_taxa[_id] = (unassignable_label, 0.0)
+
+    return obs_taxa
+
+
+def _run_command(cmd, verbose=True):
+    if verbose:
+        print("Running external command line application. This may print "
+              "messages to stdout and/or stderr.")
+        print("The command being run is below. This command cannot "
+              "be manually re-run as it will depend on temporary files that "
+              "no longer exist.")
+        print("\nCommand:", end=' ')
+        print(" ".join(cmd), end='\n\n')
+    subprocess.run(cmd, check=True)
+
+
+def _import_blast_assignments(
+        assignments, ref_taxa,
+        unassignable_label=_get_default_unassignable_label()):
+    '''import observed assignments to dict of lists.
+
+    assignments: path or list
+        Taxonomy observation map in blast format 6 or 7. Each line consists of
+        taxonomy assignments of a query sequence in tab-delimited format:
+            <query_id>    <assignment_id>   <...other columns are ignored>
+
+    ref_taxa: dict or pd.Series
+        Reference taxonomies in tab-delimited format:
+            <accession ID>  kingdom;phylum;class;order;family;genus;species
+    '''
+    obs_taxa = {}
+    lines = _open_list_or_file(assignments)
+
+    for line in lines:
+        if not line.startswith('#') or line == "":
+            i = line.split('\t')
```

Comment:
Doesn't really matter all that much, but this would be a better way to do this:

```
fields = line.split('\t') 
```

`i` is traditionally used to represent an index, so it's just a little confusing to use it this way.